### DISCUSSION TITLE
How can I know all of the arguments and environment variables that I can use to configure code-server?

### DISCUSSION THREAD

**Discussion by ghost** [coder/code-server#6726] [Q&A] [answered]
It seems there are some arguments which can be used in command line, but they are not in the output of 'code-server --help'. For example, I find someone used '--host' and '--port' in the Internet, but I can't find them in the output of '--help'. So how can I know all of the arguments that I can use in command line and the environment variables I can use, as well as their function?
Thanks a lot for answer!

**Comment by code-asher** [comment, marked as answer]
`--host` and `--port` are not listed in `--help` because they were deprecated by `--bind-addr` (example: `--bind-addr 0.0.0.0:8081`).

The only way to view hidden options right now is to look at the source: https://github.com/coder/code-server/blob/main/src/node/cli.ts#L137

**Reply by code-asher** [reply]
Environment variables do not show up either unfortunately, so for those you also have to look at the source.
